### ISSUE TITLE
How much memory is required to seed tiles?

### ISSUE THREAD

**Issue by parsispresswala** [mapserver/mapcache#191]
Hello,
I am running mapserver on 61 GB RAM and cpu of 8 core.
When I am trying to seed the tiles it is showing me following error.

I am seeding tiles with following command:
mapcache_seed -c /etc/apache2/mapcache.xml -t asia-tileset -g grid2 -n 4 -z 3,3

### 
failed to seed tile z3,x0,y0:
curl failed to request url http://13.126.201.198/cgi-bin/mapserv?VERSION=1.1.1&REQUEST=GetMap&SERVICE=WMS&STYLES=&BBOX=-20428865.584820%2c-20428865.584820%2c20428866.270399%2c20428866.270399&WIDTH=2088&HEIGHT=2088&FORMAT=image%2fpng&SRS=EPSG%3a3857&LAYERS=osm_new_admin%2cosm_new_adminocean%2cosm_new_landusages%2cosm_new_water%2cbuildings%2caeroway%2cosm_new_waterways%2cosm_new_railways%2cplanet_osm_roads%2cplanet_osm_roads_line%2cosm_new_railways1%2ccountry_line%2ccountry_line_claim%2cstate_line%2cmotorway_label%2cmainroad_label%2cminorroad_label%2cosm_new_railways_symbol%2cplace-country%2cplace-state%2cplace-city%2cplace-town%2cplace-Suburb%2cplace-village%2cplace-locality%2carea_label%2caerodrome_label%2cwaterway_label%2cwaterarea_label%2cpoi_pointlabel&MAP=%2fhome%2fubuntu%2fworldmap%2fasiamap.map : The requested URL returned error: 504 Gateway Timeout

aborting seed as 100.0% of the last 1000 requests failed

I was seeding tiles successfully when I have only two continents (asia + central america) in database.But as I insert other continents in database I am getting error. 
Can I generate tiles with same configurations?
Thanks in advance.

**Comment by jratike80** [comment]
The right forum for asking questions if the mapserver-users mailing list. Issue tracker is meant for reporting bugs.

I don't know your rendering styles but you may read all too much data for the level 0 tile that covers the whole world. You should use much simplified geometries with less vertices at that scale,  and skip for example buildings and motorway labels totally. Skipping some data may already happen for you if your styles have reasonable values for min/max scaledenominators but without seeing your mapfile it is impossible to say.

**Comment by parsispresswala** [comment]
@jratike80 I have used MAXSCALEDENOM and MINSCALEDENOM in every layer. Here is my mapfile. What should I change in it?
[http://13.126.201.198/asiamap.map](url)

**Comment by jratike80** [comment]
Is your mapfile based on the basemaps from https://github.com/mapserver/basemaps? It is a large mapfile and with a quick look it tries to prevent reading all too much data. For finding the bottle neck you should analyse what queries are really made for the PostGIS database at zoom level 0, sort them by execution time and pay attention to the slowest ones. You may find the queries from MS_ERRORFILE if you insert some DEBUG instructions into your mapfile or probably easier from PostgreSQL logs in you first turn on the logging of statements.

I do not personally render OSM data so I am not an expert on this topic. You may find Mapserver basemap users from the mailing list.

**Comment by parsispresswala** [comment]
No I haven't use github project. I have followed mapserver document from [https://mapserver.org/documentation.html](url) and generate mapfile on my own. 
what change should I do in mapfile.
Debugging mapfile MS_ERRORFILE returns following 

> message.
msConnPoolClose(): General error message. Closing connection dbname='osmdb' host='localhost' port=5432 user='parsis' password='parsis_123' sslmode=disable even though ref_count=1.
